[Feat] 하단 네비게이션(AppNav) UI 및 경로 연동 구현

AshtonSW opened the pull request:
## 🔖 PR 유형
- [x] ✨ 기능 추가
- [ ] 🐛 버그 수정
- [ ] ♻️ 리팩토링
- [ ] 🧪 테스트 코드 추가
- [ ] 📄 문서 수정
- [ ] 기타

## 📌 개요
하단 네비게이션 UI 구조를 구현하고자 `AppNav.vue`를 작성하였습니다.  
해당 컴포넌트는 주요 페이지(`/`, `/reservation`, `/fancard`, `/market`, `/mypage`)에서 사용되며,  
디자인 가이드에 따라 양 끝 여백 및 요소 간 간격을 정확히 구현하는 데 중점을 두었습니다.

초기 구현 단계에서는 전역 고정 방식으로 네비게이션을 표시하고 있습니다. 
그러나 향후 일부 페이지(예: 마켓 상세페이지, 결제 페이지 등)에서는 별도의 네비게이션 구성을 고려하고 있으므로 
추후 경로 리팩토링 예정에 있습니다.

## 🔧 작업 내용
- 하단 네비게이션(`AppNav.vue`) 컴포넌트 작성  
  - SVG 아이콘 적용 및 Active 상태 처리  
  - `w-[56px]` 고정 너비로 아이템 균등 배치  
  - `isActive()` 함수로 현재 경로 판별  
- 전체 레이아웃 구조 개선  
  - `max-w-[393px] md:max-w-[430px]` wrapper 적용  
  - 하단 네비게이션 중앙 정렬 문제 해결 (`w-[393px]` → `w-full mx-auto` 구조 변경)  
- 라우팅 경로 연결 완료  
  - `/`, `/reservation`, `/fancard`, `/market`, `/mypage` 각각 정상 연결 확인  
<img width="363" height="778" alt="image" src="https://github.com/user-attachments/assets/75627b81-b09b-4f22-9aee-f62fa8e1aaea" />


## ✅ 체크리스트
- 각 페이지별로 하단 네비게이션이 잘 렌더링되는지
- 팬카드 아이콘이 정확히 정중앙에 배치되는지 시각적으로 확인
- Active 상태의 색상 및 아이콘 전환 동작 정상 여부

## 📝 기타 참고 사항
- 향후 페이지에 따라 **다른 종류의 네비게이션 바(AppNav 대체용)**가 등장할 수 있음  
  → 예: 마켓 페이지, 결제 페이지 등에서는 별도 네비게이션이 필요함
  → 현재 구조는 전역 고정이지만, 추후 조건부 렌더링 또는 라우트 기반 분기 적용 예정

## 📎 관련 이슈
Close #8

AshtonSW committed:
[style] 화면 최대 너비 변경 및 스타일 수정

- 화면 최대 너비를 393px 에서 430px 로 변경
- 스타일 수정: app-container 클래스에 background-color, min-height, overflow-x 속성 추가

AshtonSW committed:
fix: index.html 충돌 문제 해결

AshtonSW committed:
[feat] 불필요 클래스 제거 및 네비게이션 바에 사용될 이미지 파일 추가

AshtonSW committed:
[feat] 하단 네비게이션 바(AppNav.vue) 구현 및 경로별 연결 완료

AshtonSW committed:
[fix] 네비게이션 바를 w-full + max-w로 변경하여 다양한 디바이스에서 중앙 정렬 문제 해결

AshtonSW committed:
[refactor] 불필요 코드 리팩토링

DaOpen commented:
확인 완료했습니다! 수고하셨습니다 👍

icegosimperson commented:
확인했습니다 👍

yonseeee commented:
컴포넌트 분리가 잘 된 좋은 코드인 것 같습니다
이제 프론트까지 접수하시다니 어디까지 성장하실 지 기대가 됩니다
LGTM:)

joojinhyeok commented:
확인했습니다!!

khanz0613 commented:
고생 많으셨습니다!!
다만 AppNav.vue 에서 
const isActive = (path) => route.path === path
이 코드는 정확히 해당 페이지에 있을 경우에만 활성화 된다고 합니다! 그래서 해당 페이지의 하위 페이지로 들어갈 경우에는 네비게이션 바가 까맣게 되지 않는다고 하네용
const isActive = (path) => route.path.startsWith(path) 
로 수정하시면 해결될  것으로 보입니다.
공통 컴포넌트 만들어주셔서 감사합니다!

AshtonSW commented:
> 고생 많으셨습니다!! 다만 AppNav.vue 에서 const isActive = (path) => route.path === path 이 코드는 정확히 해당 페이지에 있을 경우에만 활성화 된다고 합니다! 그래서 해당 페이지의 하위 페이지로 들어갈 경우에는 네비게이션 바가 까맣게 되지 않는다고 하네용 const isActive = (path) => route.path.startsWith(path) 로 수정하시면 해결될 것으로 보입니다. 공통 컴포넌트 만들어주셔서 감사합니다!



감사합니다! 이런 경우를 고려 못 했네요! :D 수정 완료입니다!
홈 경로일 때랑 충돌하는 문제가 있어서 조건문 달아서 홈경로 아닐 때만 StartsWith 적용되도록 합니다~!
```
const isActive = (path) => {
  if (path === '/') {
    return route.path === '/'
  }
  return route.path.startsWith(path)
}
```

AshtonSW commented:
> > 고생 많으셨습니다!! 다만 AppNav.vue 에서 const isActive = (path) => route.path === path 이 코드는 정확히 해당 페이지에 있을 경우에만 활성화 된다고 합니다! 그래서 해당 페이지의 하위 페이지로 들어갈 경우에는 네비게이션 바가 까맣게 되지 않는다고 하네용 const isActive = (path) => route.path.startsWith(path) 로 수정하시면 해결될 것으로 보입니다. 공통 컴포넌트 만들어주셔서 감사합니다!
> 
> 감사합니다! 이런 경우를 고려 못 했네요! :D 수정 완료입니다! 홈 경로일 때랑 충돌하는 문제가 있어서 조건문 달아서 홈경로 아닐 때만 StartsWith 적용되도록 합니다~!
> 
> ```
> const isActive = (path) => {
>   if (path === '/') {
>     return route.path === '/'
>   }
>   return route.path.startsWith(path)
> }
> ```

<img width="368" height="789" alt="image" src="https://github.com/user-attachments/assets/93955107-6355-4a6d-bf8b-343cc4b703db" />

AshtonSW committed:
[fix] 하단 네비게이션 isActive 로직 개선 (startsWith 적용)

- route.path === path 방식에서 route.path.startsWith(path) 로 변경
- 상세 경로(`/reservation/detail` 등)에서도 상위 메뉴가 active 되도록 처리

khanz0613 commented:
> > > 고생 많으셨습니다!! 다만 AppNav.vue 에서 const isActive = (path) => route.path === path 이 코드는 정확히 해당 페이지에 있을 경우에만 활성화 된다고 합니다! 그래서 해당 페이지의 하위 페이지로 들어갈 경우에는 네비게이션 바가 까맣게 되지 않는다고 하네용 const isActive = (path) => route.path.startsWith(path) 로 수정하시면 해결될 것으로 보입니다. 공통 컴포넌트 만들어주셔서 감사합니다!
> > 
> > 
> > 감사합니다! 이런 경우를 고려 못 했네요! :D 수정 완료입니다! 홈 경로일 때랑 충돌하는 문제가 있어서 조건문 달아서 홈경로 아닐 때만 StartsWith 적용되도록 합니다~!
> > ```
> > const isActive = (path) => {
> >   if (path === '/') {
> >     return route.path === '/'
> >   }
> >   return route.path.startsWith(path)
> > }
> > ```
> 
> <img alt="image" width="368" height="789" src="https://private-user-images.githubusercontent.com/92350806/470048363-93955107-6355-4a6d-bf8b-343cc4b703db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzMTgzOTAsIm5iZiI6MTc1MzMxODA5MCwicGF0aCI6Ii85MjM1MDgwNi80NzAwNDgzNjMtOTM5NTUxMDctNjM1NS00YTZkLWJmOGItMzQzY2M0YjcwM2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDAwNDgxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNmE4ZGJhMjlkZjYyZDM0MjI1YmNjYzQ0N2VkNTM2YjA3NmYwODMwYzlhZDI4ODY3MDFkOWY4MTVmOWYzNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jzxgk9A-vdtRxDwmH0wtgI4fC5vcl-QWvSBSsU9O5l8">

고생 많으셨습니다!
이 코드는 굉장히 높게 평가... 👍
